Add a script to translate commcare with openai

AmitPhulera opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->
A prototype of using LLMs to make translations in HQ
## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->


### Migrations
<!-- Delete this section if the PR does not contain any migrations -->
<!-- https://commcare-hq.readthedocs.io/migrations_in_practice.html -->
- [ ] The migrations in this code can be safely applied first independently of the code

<!-- Please link to any past code changes that are coordinated with this migration -->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [ ] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [ ] Risk label is set correctly
- [ ] The set of people pinged as reviewers is appropriate for the level of risk of the change

AmitPhulera committed:
spanish translations

AmitPhulera committed:
Merge remote-tracking branch 'origin/master' into ap/es-translations

AmitPhulera committed:
french translation

AmitPhulera committed:
portugal translation

AmitPhulera committed:
translate with openai management command

snopoke commented on corehq/apps/hqwebapp/management/commands/translate_with_openai.py:
```diff
@@ -0,0 +1,156 @@
+	  ],
+	  temperature=0.2,
+	)
+	return json.loads(completion.choices[0].message.content)
```
I think it would be useful to structured outputs to get a better chance of it producing the right result: https://platform.openai.com/docs/guides/structured-outputs

Either using `response_format` or using function calling

AmitPhulera committed:
fix french translations

AmitPhulera committed:
lint fixes

AmitPhulera committed:
Merge remote-tracking branch 'origin/master' into ap/es-translations

AmitPhulera committed:
more french translation issue fixes

AmitPhulera committed:
add japanese

AmitPhulera committed:
merge translations from master

AmitPhulera committed:
add jpn in settings.py in languages

AmitPhulera committed:
add japanese to update_django_locales

AmitPhulera committed:
merge translations from master. Contains some errors in compiling messages. Will fix later

AmitPhulera committed:
fixes warnings in the translation generation

./corehq/apps/reports/views.py:806: warning: 'msgid' format string with unnamed arguments cannot be properly localized:
                                             The translator cannot reorder the arguments.
                                             Please consider using a format string with named argument

AmitPhulera committed:
fixes warning - Empty msgid.  It is reserved by GNU gettext:
                                                 gettext() returns the header entry with
                                                 meta information, not the empty string.

AmitPhulera committed:
ask opeai for a json_object in response

AmitPhulera committed:
add todos

AmitPhulera committed:
parallelize api calls with gevent

AmitPhulera committed:
compile po files after generation

AmitPhulera committed:
update transaltion string iteration as per new format and make the function more fault tolerant

AmitPhulera committed:
let management command use gevent

AmitPhulera committed:
added deu and hin
